File: app/src/main/java/com/shevaalex/android/rickmortydatabase/models/ApiPageModel.java
package com.shevaalex.android.rickmortydatabase.models;

import androidx.annotation.NonNull;

import com.google.gson.annotations.SerializedName;
import com.shevaalex.android.rickmortydatabase.source.network.net_utils.ApiConstants;

public abstract class ApiPageModel {
    @SerializedName(ApiConstants.INFO)
    private ApiPageInfoModel apiPageInfoModel;

    public ApiPageInfoModel getApiPageInfoModel() {
        return apiPageInfoModel;
    }

    @NonNull
    @Override
    public String toString() {
        return "ApiPageModel{" +
                "apiPageInfoModel=" + apiPageInfoModel +
                '}';
    }
}

File: app/src/main/java/com/shevaalex/android/rickmortydatabase/models/ApiPageModel.kt
package com.shevaalex.android.rickmortydatabase.models

import com.google.gson.annotations.SerializedName
import com.shevaalex.android.rickmortydatabase.utils.networking.ApiConstants

abstract class ApiPageModel(
        @SerializedName(ApiConstants.INFO)
        val apiPageInfoModel: ApiPageInfoModel
) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is ApiPageModel) return false

        if (apiPageInfoModel != other.apiPageInfoModel) return false

        return true
    }

    override fun hashCode(): Int {
        return apiPageInfoModel.hashCode()
    }

    override fun toString(): String {
        return "ApiPageModel(apiPageInfoModel=$apiPageInfoModel)"
    }
}